zk/stages/stage_sequence_execute.go

```diff
@@ -120,6 +120,7 @@ func sequencingBatchStep(
 	batchState := newBatchState(forkId, batchNumberForStateInitialization, executionAt+1, cfg.zk.UseExecutors(), cfg.zk.L1SyncStartBlock > 0, cfg.txPool, resequenceBatchJob)
 	blockDataSizeChecker := NewBlockDataChecker(cfg.zk.ShouldCountersBeUnlimited(batchState.isL1Recovery()))
 	streamWriter := newSequencerBatchStreamWriter(batchContext, batchState)
+	normalcyGasPool := new(core.GasPool).AddGas(transactionGasLimit) // used in normalcy mode per block
```

Comment:
This is a little confusing naming wise.  We send this into every transaction run whether it is normalcy or not.  Maybe something like `ethBlockGasPool` and a comment mentioning that this will only be used in the case of normalcy mode otherwise it will be replaced on a per transaction basis.

Also, does this line not need to be inside the block loop?  At the moment this is created only once per batch?